provider/netperf_test.py

```diff
@@ -59,11 +59,22 @@ def netperf_stress(test, params, vm):
             logging.info("Netperf test start successfully.")
         else:
             test.error("Can not start netperf client.")
-        utils_misc.wait_for(lambda: not n_client.is_netperf_running(),
-                            netperf_test_duration, 0, 5,
-                            "Wait netperf test finish %ss" % netperf_test_duration)
-        time.sleep(5)
+        start_time = time.time()
+        utils_misc.wait_for(lambda: not
+                            n_client.is_netperf_running(),
+                            test_duration, 0, 5,
+                            "Wait netperf test finish %ss" % test_duration)
+        stop_time = time.time()
+        run_time = stop_time - start_time
+        if n_client.is_netperf_running():
+            test.fail("netperf still running, it hangs")
+        elif test_duration - 5 <= run_time <= test_duration + 2:
+            logging.info("netperf runs successfully")
+        elif test_duration + 2 < run_time:
+            test.fail("neterf runing timeout,executed %ss" % run_time)
+        else:
+            test.fail("netperf terminated unexpectedly,executed %ss" % run_time)
```

Comment:
```suggestion
        execution_time = test_duration + 3
        utils_misc.wait_for(lambda: not
                            n_client.is_netperf_running(),
                            execution_time, 0, 5,
                            "Wait netperf test finish %ss" % test_duration)
        stop_time = time.time()
        run_time = stop_time - start_time
        if n_client.is_netperf_running():
            test.fail("netperf still running")
        elif test_duration - 5 <= run_time:
            logging.info("netperf runs successfully")
        else:
            test.fail("netperf terminated unexpectedly,executed %ss" % run_time)
```

Reply:
Hi @PaulYuuu Thanks for you help, I have updated it. Please help me review it again.Thanks a lot.  According to the test result summary, windows sometimes takes longer, so I extended the offset time to 5 seconds. If the test result needs to be provided, please let me know.